### PR TITLE
Qualification tool: Parse expressions in ProjectExec

### PR DESCRIPTION
**Pull request by nartal1**
This contributes to https://github.com/NVIDIA/spark-rapids/issues/5617 .
In this PR we parse expressions which are in ProjectExec. Project may have aliases so made sure we are taking that into account. Tested with several expressions locally but added only 1 test each for supporting and non-supporting case. Supporting case pretty much covers all cases where the ProjectExec string needs to be parsed.
There was a bug in FilterExec parser where we are not taking into account if the predicate was `+`. Updated that as well.

### CODE REVIEW

**Comment by nartal1** [comment]
build

**Comment by tgravescs** [comment]
If you haven't yet we should test against the NDS event logs

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala [hunk 1]
```diff
@@ -268,6 +268,25 @@ object SQLPlanParser extends Logging {
     funcName
   }
 
+  def parseProjectExpressions(exprStr: String): Array[String] = {
+    val parsedExpressions = ArrayBuffer[String]()
+    // remove the alias names before parsing
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala:273]: can you add an example in the comment

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala [hunk 2]
```diff
@@ -268,6 +268,25 @@ object SQLPlanParser extends Logging {
...
+    val pattern = """(AS) ([(\w# )]+)""".r
+    // This is to split multiple column names in Project. Project may have a function on a column.
+    // This will contain array of columns names specified in ProjectExec.
+    val paranRemoved = pattern.replaceAllIn(exprStr.replace("),", "::"), "")
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala:277]: lets start putting more description on these so when we come back to them we can understand.... put example as well.

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala [hunk 3]
```diff
@@ -556,4 +557,59 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
...
   }
+
+  test("Expressions supported in ProjectExec") {
+    TrampolineUtil.withTempDir { outputLoc =>
+      TrampolineUtil.withTempDir { eventLogDir =>
+        val (eventLog, _) = ToolTestUtils.generateEventLog(eventLogDir,
+          "ProjectExprsSupported") { spark =>
+          import spark.implicits._
+          val df1 = Seq(9.9, 10.2, 11.6, 12.5).toDF("value")
+          df1.write.parquet(s"$outputLoc/testtext")
```
Comment (tgravescs) [inline tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala:568]: put parquet in name of directory

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala [hunk 4]
```diff
@@ -556,4 +557,59 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
...
+  }
+
+  test("Expressions not supported in ProjectExec") {
+    TrampolineUtil.withTempDir { outputLoc =>
+      TrampolineUtil.withTempDir { eventLogDir =>
+        val (eventLog, _) = ToolTestUtils.generateEventLog(eventLogDir,
+          "ProjectExprsNotSupported") { spark =>
+          import spark.implicits._
+          val df1 = spark.sparkContext.parallelize(List(10, 20, 30, 40)).toDF
+          df1.write.parquet(s"$outputLoc/testtext")
```
Comment (tgravescs) [inline tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala:596]: same here put parquet inname for outputloc

**Comment by nartal1** [comment]
> If you haven't yet we should test against the NDS event logs

I tested this against NDS event logs and surprisingly we don't have any unsupported exprs in any of the ProjectExec. I created some custom eventlogs with unsupported exprs to be sure that the testing is done correctly.

**Comment by nartal1** [comment]
build

**Pull request merged**